### PR TITLE
compresses sequents with the same body in preprocessing

### PR DESCRIPTION
**Pull request by salmans**

### CODE REVIEW

File: src/formula/transform.rs [hunk 1]
```diff
@@ -685,6 +686,81 @@ impl Formula {
...
+                    variables: variables.to_vec(),
+                    formula: Box::new(distribute_and(formula)),
+                },
+                _ => panic!("Internal Error: Expecting a formula in negation normal form.") // already NNF
+            }
+        }
+
+        // The following eliminates the existential quantifiers of the input formula.
+        fn remove_forall(formula: Formula) -> Formula {
+            if let Forall { variables: _, formula } = formula {
```
Comment (KiChjang) [inline src/formula/transform.rs]:
```suggestion
            if let Forall { formula, .. } = formula {
```

**Comment by KiChjang** [comment]
Looks like I left some on #30 as well, didn't know that I'm supposed to look at this one first...

File: src/formula/transform.rs (hunk 1 above)
Comment (salmans) [reply src/formula/transform.rs]: Nice one!

**Comment by salmans** [comment]
@KiChjang no worries - the order doesn't matter. I appreciate your feedback in any order!

**Pull request closed without merging**